src/methods.jl

```diff
@@ -218,14 +218,14 @@ julia> fdm(central_fdm(2, 1), exp, 0, Val(true))
 function fdm(
     m::M,
     f,
-    x,
+    x::T,
     ::Val{true};
     condition=DEFAULT_CONDITION,
     bound=_estimate_bound(f(x), condition),
-    eps=(Base.eps(float(bound)) + TINY),
+    eps=add_tiny(Base.eps(T(bound))),
     adapt=m.history.adapt,
-    max_step=0.1,
-) where M<:FiniteDifferenceMethod
+    max_step=T(0.1),
+) where {T<:AbstractFloat, M<:FiniteDifferenceMethod}
```

Comment:
Why not `Real`?
```suggestion
) where {T<:Real, M<:FiniteDifferenceMethod}
```

Reply:
See line 280 (ish). If an `Integer` is provided because someone types `1` rather than `1.0` then we need control over what type it converts to. Longer term I think we probably want to disallow `Integer` inputs altogether, but that's not something to be addressed in this PR.